File: my-folder/0150-evaluate-reverse-polish-notation/solution.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        # use a stack to store the digits when a operator comes do the operation and push result into the stack
        stack = []
        operators = set(["+", "-", "*", "/"])
        for token in tokens:
            if token not in operators:
                stack.append(int(token))
            else:
                if len(stack) >= 2:
                    digit2 = stack.pop()
                    digit1 = stack.pop()
                    print(digit1, digit2)
                    if token == "+":
                        stack.append(digit1 + digit2)
                    elif token == "-":
                        stack.append(digit1 - digit2)
                    elif token == "*":
                        stack.append(digit1 * digit2)
                    elif token == "/":
                        stack.append(int(float(digit1) / digit2))
        return stack[-1]

File: my-folder/0150-evaluate-reverse-polish-notation/solution.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();
        for(String token : tokens){
            if(!token.equals("*") && !token.equals("+")  && !token.equals("-")  && !token.equals("/")){
                stack.push(Integer.parseInt(token));
            }
            else{
                int res = 0;
                int op1 = stack.pop();
                int op2 = stack.pop();
                if(token.equals("+")){
                    res += op1 + op2;
                }
                else if(token.equals("-")){
                    res += op2 - op1;
                }
                else if(token.equals("*")){
                    res += op1 * op2;
                }
                else if(token.equals("/")){
                    res += op2 / op1;
                }
                stack.push(res);
            }
        }

        return stack.pop();
    }
}
